### ISSUE TITLE
`getline` corrupts string

### ISSUE THREAD

**Issue by xonixx** [onetrueawk/awk#133]
```
$ echo 'a' | ./soft/gawk51 'BEGIN { getline l; getline l; print (s=substr(l,1,10)) " len=" length(s) }'
a len=1

$ echo 'a' | ./soft/mawk134 'BEGIN { getline l; getline l; print (s=substr(l,1,10)) " len=" length(s) }'
a len=1

$ echo 'a' | ./soft/goawk1.8.1 'BEGIN { getline l; getline l; print (s=substr(l,1,10)) " len=" length(s) }'
a len=1

$ echo 'a' | ./soft/bwk 'BEGIN { getline l; getline l; print (s=substr(l,1,10)) " len=" length(s) }'
xK$� len=6
```

`bwk` is one true awk. Obviously the second `getline` does something wrong here.

**Comment by plan9** [comment]
thank you for the find. I'll investigate.

**Comment by millert** [comment]
I debugged the uninitialized access with valgrind, the resulting fix is #134

**Comment by plan9** [comment]
thank you Todd.
